### PR TITLE
Distressed position state

### LINKED ISSUE #7846: Add "distressed" status for traders
link to [PR 1622](https://github.com/vegaprotocol/specs/pull/1622)

Add "distressed" status for traders, new AC as below:

The status field will be set to DISTRESSED if a trader was distressed based on the margin requirements for their worst possible long/short and they do not have active orders to be closed, however the book currently does not have enough volume to close them out, and will close them out when there is enough volume.(<a name="0007-POSN-017" href="#0007-POSN-017">0007-POSN-017</a>)

### PR DESCRIPTION
**Pull request by EVODelavega**
Closes #7846 

When a trader is distressed, but the network order can't uncross, emit an event so we can mark such parties as being distressed, despite having an open position.

### CODE REVIEW

File: protos/sources/vega/events/v1/events.proto [hunk 1]
```diff
@@ -719,6 +732,8 @@ enum BusEventType {
   BUS_EVENT_TYPE_DISTRESSED_ORDERS_CLOSED = 56;
   // Event indicating parties had orders closed because they were distressed, but were not closed out.
   BUS_EVENT_TYPE_EXPIRED_ORDERS = 57;
+  // Event indicating parties have become, or were distressed but still have an active position.
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:735]:
```suggestion
  // Event indicating parties have become, or were, distressed but still have an active position.
```

File: protos/sources/vega/events/v1/events.proto [hunk 2]
```diff
@@ -429,6 +429,19 @@ message DistressedOrders {
...
 }
 
+// A distressed positions event contains the party IDs for all parties that were distressed, had their orders closed but because of insufficient volume on the book
+// could not be fully closed out. These parties are distressed, but still hold an active position on the book as a result. Once enough volume is on the book to close
+// them out, a SettleDistressed event will be sent. In case they manage to reduce their position, or meet the margin requirements, this status will be updated.
+// Parties that are no longer distressed but active will be listed in the safe_parties field.
+message DistressedPositions {
+  // Market identifier for the event
+  string market_id = 1;
+  // Slice of Party identifiers (public keys) that are distressed but still have open volume
```
Comment (candida-d) [inline protos/sources/vega/events/v1/events.proto:439]:
```suggestion
  // Slice of party identifiers (public keys) that are distressed but still have open volume
```

**Review by candida-d — COMMENTED** [review]
Very minor documentation suggestions

**Review by Jiajia-Cui — DISMISSED** [review]
the feature tests look good to me

**Review by Jiajia-Cui — DISMISSED** [review]
feature tests look good

**Review by candida-d — DISMISSED** [review]

**Review by ze97286 — DISMISSED** [review]

**Review by jeremyletang — DISMISSED** [review]

**Review by candida-d — DISMISSED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**